shared/ClrFacade/InternalRDotNetDataConverter.cs

```diff
@@ -324,13 +324,22 @@ private GenericVector convertDictionary<T>(object obj)
       return values.AsList();
    }
 
-   private SymbolicExpression convertAll(IReadOnlyList<object> objects, Func<object, SymbolicExpression> converter = null)
+   private SymbolicExpression convertAll(IReadOnlyList<object> objects)
    {
       var sexpArray = new SymbolicExpression[objects.Count];
 
       for (var i = 0; i < objects.Count; i++)
-         sexpArray[i] = converter == null ? convertToSexp(objects[i]) : converter(objects[i]);
-
+      {
+         if (objects[i] == null)
+         {
+            sexpArray[i] = _engine.NilValue;
+         }
+         else
+         {
+            var elementConverter = tryGetConverter(objects[i].GetType());
```

Comment:
this converter lookup is now done _per-element_ instead of _per-array_

what is involved in a converter lookup? Is it expensive?

Reply:
Here's the code for it

```
   private Func<object, SymbolicExpression> tryGetConverter(Type t)
   {
      return _converterFunctions.GetValueOrDefault(t);
   }
```

_converterFunctions is a Microsoft Dictionary object

```
   private readonly Dictionary<Type, Func<object, SymbolicExpression>> _converterFunctions;
```

`GetValueOrDefault` is an extension method from .NET. I doubt we would be able to write anything faster. If we profile and find a big difference then we could implement a check if the key exists in the dictionary first. But I would be shocked if that was not exactly how this is already done.

Reply:
oh, the way to test if this has a big impact is to send a big array of objects that have no converter in the dictionary.